solarforecastarbiter/reference_forecasts/utils.py

```diff
@@ -191,30 +192,32 @@ def _dayahead_start_end(run_time):
     until end of day. So, forecast *never* uses obs > 24 hr old at each
     valid time. Arguably too much for a reference forecast.
     """
-
-    data_end = run_time.floor('1d')
+    # data_end = last forecast time for next issue of run - 1 day
+    data_end = issue_time + forecast.lead_time_to_start + forecast.run_length
+    while data_end > run_time:
+        data_end -= pd.Timedelta('1d')
```

Comment:
Is this meant to ensure that 1. `data_end` is always less than `run_time` and 2. `data_end.time()` doesn't change?

Can the same thing can be accomplished more efficiently with something like

```python
if data_end > run_time:
    day_offset = (data_end - run_time).days
    data_end -= pd.Timedelta(days=day_offset)
```

Reply:
maybe I'm misunderstanding something since I would have thought the same requirement would be added to `_weekahead_start_end`.

Reply:
Actually, I think what's important is that `data_end < issue_time` and `data[i].time() == forecast[i].time()` (so agreed on 2). I've updated to enforce this, although it can lead to quirks like a 23h ahead forecast using more recent data than a 1h ahead. 

As for weekahead, there it's important that the day of the week along with time is maintained. So just subtracting the week from the forecast start and assuming run_time is after data end should be ok. If a set of parameters always violates that, the parameters should be changed.